modules/social_features/social_group/src/Plugin/EntityReferenceSelection/SocialGroupSelection.php

```diff
@@ -42,27 +43,31 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $all_group_types = $storage->getQuery()->execute();
     }
 
-    $excluded_group_types = [];
+    if (is_array($all_group_types)) {
+      $excluded_group_types = [];
+      foreach ($all_group_types as $group_type_id) {
+        /** @var \Drupal\group\Entity\GroupTypeInterface $group_type */
+        $group_type = $storage->load($group_type_id);
 
-    foreach ($all_group_types as $group_type_id) {
-      /** @var \Drupal\group\Entity\GroupTypeInterface $group_type */
-      $group_type = $storage->load($group_type_id);
-
-      if (!$group_type->hasContentPlugin($plugin_id)) {
-        $excluded_group_types[] = $group_type_id;
+        if (!$group_type->hasContentPlugin($plugin_id)) {
+          $excluded_group_types[] = $group_type_id;
+        }
       }
-    }
 
-    if ($excluded_group_types) {
-      $diff = array_diff($all_group_types, $excluded_group_types);
-      if (!empty($diff)) {
-        $bundle = $this->entityTypeManager->getDefinition($configuration['target_type'])->getKey('bundle');
-        if (is_string($bundle)) {
-          $query->condition(
-            $bundle,
-            $diff,
-            'IN'
-          );
+      if ($excluded_group_types) {
```

Comment:
@navneet0693 check if you can reduce the amount of nested if statements, perhaps also move some logic to separated methods.

Reply:
Okay.